components/camera/fake/camera.go

```diff
@@ -170,59 +168,60 @@ func fakeModel(width, height int) (*transform.PinholeCameraModel, int, int) {
 // Camera is a fake camera that always returns the same image.
 type Camera struct {
 	generic.Echo
-	Name   string
-	Model  *transform.PinholeCameraModel
-	Width  int
-	Height int
+	Name            string
+	Model           *transform.PinholeCameraModel
+	Width           int
+	Height          int
+	cacheImage      *image.RGBA
+	cachePointCloud *pointcloud.PointCloud
 }
 
-// Read always returns the same image of a chess board.
+// Read always returns the same image of a yellow to blue gradient.
 func (c *Camera) Read(ctx context.Context) (image.Image, func(), error) {
-	img, err := c.getColorImage()
-	return img, func() {}, err
-}
-
-// NextPointCloud always returns a pointcloud of the chess board.
-func (c *Camera) NextPointCloud(ctx context.Context) (pointcloud.PointCloud, error) {
-	img, err := c.getColorImage()
-	if err != nil {
-		return nil, err
+	if c.cacheImage != nil {
+		return c.cacheImage, func() {}, nil
 	}
-	dm, err := c.getDepthImage(ctx)
-	if err != nil {
-		return nil, err
+	width := float64(c.Width)
+	height := float64(c.Height)
+	img := image.NewRGBA(image.Rect(0, 0, c.Width, c.Height))
+
+	totalDist := math.Sqrt(math.Pow(0-width, 2) + math.Pow(0-height, 2))
+
+	var x, y float64
+	for x = 0; x < width; x++ {
+		for y = 0; y < height; y++ {
+			dist := math.Sqrt(math.Pow(0-x, 2) + math.Pow(0-y, 2))
+			dist /= totalDist
+			thisColor := color.RGBA{uint8(255 - (255 * dist)), uint8(255 - (255 * dist)), uint8(0 + (255 * dist)), 255}
+			img.Set(int(x), int(y), thisColor)
+		}
 	}
-	return c.Model.RGBDToPointCloud(img, dm)
-}
 
-// getColorImage always returns the same color image of a chess board.
-func (c *Camera) getColorImage() (*rimage.Image, error) {
-	img, err := rimage.NewImageFromFile(artifact.MustPath("rimage/board2.png"))
-	if err != nil {
-		return nil, err
-	}
-	if c.Height == initialHeight && c.Width == initialWidth {
-		return img, nil
-	}
-	dst := image.NewRGBA(image.Rect(0, 0, c.Width, c.Height))
-	draw.NearestNeighbor.Scale(dst, dst.Bounds(), img, img.Bounds(), draw.Over, nil)
-	return rimage.ConvertImage(dst), nil
+	return rimage.ConvertImage(img), func() {}, nil
```

Comment:
Do you set the img to cacheImage anywhere?

Reply:
wow yeah great point